drivers/i2c/i2c_ll_stm32_v1.c

```diff
@@ -294,39 +300,43 @@ s32_t stm32_i2c_msg_read(struct device *dev, struct i2c_msg *msg,
 #else
 
 s32_t stm32_i2c_msg_write(struct device *dev, struct i2c_msg *msg,
-			  u32_t flags, u16_t saddr)
+			  u8_t *next_msg_flags, u16_t saddr)
 {
 	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
 	struct i2c_stm32_data *data = DEV_DATA(dev);
 	I2C_TypeDef *i2c = cfg->i2c;
 	u32_t len = msg->len;
 	u8_t *buf = msg->buf;
 
+	ARG_UNUSED(next_msg_flags);
+
 	LL_I2C_AcknowledgeNextData(i2c, LL_I2C_ACK);
-	LL_I2C_GenerateStartCondition(i2c);
-	while (!LL_I2C_IsActiveFlag_SB(i2c)) {
-		;
-	}
+	if (msg->flags & I2C_MSG_RESTART) {
```

Comment:
Would you add empty lines between logical sections?
This whole function has gotten hard to read

Reply:
done